.github/workflows/build.yml

```diff
@@ -81,33 +54,72 @@ jobs:
           GITHUB_REF_NAME_NORMALIZED=$(echo ${{github.ref_name}} | tr / -)
           echo "GITHUB_REF_NAME_NORMALIZED=${GITHUB_REF_NAME_NORMALIZED}" >> $GITHUB_ENV
 
+          DOCKERIZE=${{ contains(fromJSON('["master", "staging", "production"]'), github.ref_name) || startsWith(github.event.head_commit.message, 'dockerize')}}
+          echo "DOCKERIZE=${DOCKERIZE}" >> $GITHUB_ENV
       - name: Setup Node
         uses: actions/setup-node@v3
         with:
-          node-version: "${{ env.NODE_VERSION }}"
-          cache: "yarn"
+          node-version-file: package.json
+          # Caching yarn dir & running yarn install is too slow
+          # Instead, we aggressively cache node_modules below to avoid calling install
 
       - name: Get cached node modules
         id: cache
         uses: actions/cache@v3
         with:
           path: |
             **/node_modules
-          key: ${{ env.DEPENDENCIES_CACHE }}-${{ hashFiles('yarn.lock') }}
+          key: node_modules-${{ runner.arch }}-${{ env.NODE_VERSION }}-${{ hashFiles('yarn.lock') }}
```

Comment:
caching node_modules is dangerous. we must be sure the arch & node version are the same otherwise they won't work.